### DISCUSSION TITLE
Dependency download issues during build

### DISCUSSION THREAD

**Discussion by reback00** [orbitalquark/textadept#78] [Q&A] [answered]
Hi. I maintain Textadept package for Void Linux.

Recently I've had [some issues](https://github.com/void-linux/void-packages/pull/29666#issuecomment-803646589) regarding cdk download during build from invisible-mirror.net. It seems like they have some DDoS protection or something that stops it from downloading. This causes the build to fail.

Is there any solution to this? Can Textadept mirror deps?

**Comment by orbitalquark** [comment, marked as answer]
Mirroring capability is not built into the makefile, but you can download the dependencies in https://github.com/orbitalquark/textadept-build and copy them to your *src/* directory before running `make deps`. The nightly build script does this instead of pulling from 3rd-party servers.

**Reply by reback00** [reply]
Seems like the answer I was looking for. I was hoping for a mirror maintained by the project itself so that downloads don't fail like that.

One question, how long will the downloads last? I mean I think a better way would be to divide them into directories with dep name then year then files. e.g. `cdk/2020/cdk-5.0-20200923.tgz`. The way it's right now would be a mess in a year or two and you guys would probably think of cleaning up. I wanted to create an issue, but it's disabled in the project.

**Comment by orbitalquark** [comment]
I don't anticipate culling the downloads for a while since users may want to build older versions of Textadept. Dependencies aren't updated very often either, so categorizing and filing them does not seem necessary to me. Thanks for bringing it up though.
